cmd/mount_zfs/mount_zfs.c

```diff
@@ -275,7 +268,10 @@ main(int argc, char **argv)
 		return (MOUNT_USAGE);
 	}
 
-	zfs_adjust_mount_options(zhp, mntpoint, mntopts, mtabopt);
+	if (!zfsutil || sloppy ||
+	    libzfs_envvar_is_set("ZFS_MOUNT_HELPER")) {
```

Comment:
Why check the `ZFS_MOUNT_HELPER` environment variable here and below?

Reply:
To avoid an infinite loop. The flow:
- `zfs_mount_at()` invokes `do_mount`
- `do_mount` invokes `/bin/mount` if `ZFS_MOUNT_HELPER` is set
- `/bin/mount` `execve()s` `mount.zfs`
- `mount.zfs` invokes `zfs_mount_at()`

This edge case was caught by the regression suite.

https://github.com/openzfs/zfs/blob/f2c5bc150e609a78185ea63c84fce7718f56e28a/lib/libzfs/os/linux/libzfs_mount_os.c#L328-L350

Reply:
Ahh, I see.  That makes sense, thanks.